mergify_engine/rules/__init__.py

```diff
@@ -95,35 +160,79 @@ async def eval(self, obj: filter.GetAttrObjectT) -> bool:
         self.match = await filter.Filter({"and": self.conditions})(obj)
         return self.match
 
-    def __getitem__(self, idx: int) -> RuleCondition:
-        return self.conditions[idx]
+    def iter_root_rule_conditions(
```

Comment:
IIRC this is temporary until we support `check-` everywhere?

Reply:
yes